### PR TITLE
84 Add tests for removing duplicate keys

### PR DESCRIPTION
**Pull request by qnikst**
PR adds simple test and introduces a logic to deduplicate keys.

### CODE REVIEW

File: scls-format/src/Cardano/SCLS/Internal/Serializer/Reference/Dump.hs [hunk 1]
```diff
@@ -108,6 +109,23 @@ dumpToHandle handle hdr orderedStream = do
...
+  initialize (Return r) = Return r
+  initialize (Effect e) = Effect (e >>= \s -> return (initialize s))
+  initialize (Step (x :> rest)) = S.yield x >> go (getKey x) rest
+  go _ (Return r) = return r
+  go p (Effect e) = Effect (e >>= \s -> return (go p s))
+  go p (Step (x :> rest)) =
+    let currentKey = getKey x
+     in if p == currentKey
+          then go p rest
+          else S.yield x >> go currentKey rest
```
[the remark this answers is not in the document]
Comment (qnikst) [reply scls-format/src/Cardano/SCLS/Internal/Serializer/Reference/Dump.hs]: o_O

File: scls-util/scls-util.cabal [hunk 2]
```diff
@@ -42,6 +42,10 @@ executable scls-util
   import: warnings
   hs-source-dirs: app
   main-is: Main.hs
+  ghc-options:
+    -O2
+    -rtsopts
```
Comment (joaosreis) [inline scls-util/scls-util.cabal]: I'm assuming this change is unrelated. Right?

**Review by joaosreis — APPROVED** [review]
LGTM, just double check if the change in `scls-util.cabal` is to be included.

**Comment by qnikst** [comment]
Fixed the non-related changes and rebased atop of the API change
